### PR TITLE
Upgrade bazel

### PR DESCRIPTION
**Pull request by stepancheg**
Something is broken in the old Bazel; just upgrade it.

### CODE REVIEW

**Comment by stepancheg** [comment]
Failing CI job is here https://github.com/bazelbuild/starlark/pull/99 for example; hope this is going to fix it.

**Comment by stepancheg** [comment]
CI still fails, but at least it fails at Java tests, not at the compilation of Bazel.

File: test_suite/BUILD [hunk 1]
```diff
@@ -20,7 +20,7 @@ load("@rules_python//python:defs.bzl", "py_test")
         )
         for test_file in glob(["testdata/java/*", "testdata/go/*", "testdata/rust/*"])
     ]
-    for impl, binary_rule in [("java", "@io_bazel//src/main/java/com/google/devtools/starlark:Starlark"),
+    for impl, binary_rule in [("java", "@io_bazel//src/main/java/net/starlark/java/cmd:Starlark"),
```
Comment (laurentlb) [inline test_suite/BUILD:23]: Also update https://github.com/bazelbuild/starlark/blob/master/test_suite/testenv.py

**Comment by stepancheg** [comment]
Done, also fixed the tests (except one, broken in Java Bazel, commented out).

**Pull request merged**